### PR TITLE
hydra-check: 1.2.0 -> 1.3.4

### PR DESCRIPTION
**Pull request by Artturin**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 169799 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>hydra-check (python310Packages.hydra-check)</li>  </ul></details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -17724,7 +17724,7 @@ with pkgs;
 
   hydraAntLogger = callPackage ../development/libraries/java/hydra-ant-logger { };
 
-  hydra-check = with python3.pkgs; toPythonApplication hydra-check;
+  hydra-check = with python310.pkgs; toPythonApplication hydra-check;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:17727]: If it still works on 3.9 we shouldn't bump this.

File: pkgs/development/python-modules/hydra-check/default.nix [hunk 2]
```diff
@@ -2,46 +2,39 @@
...
     repo = pname;
-    rev = version;
-    sha256 = "EegoQ8qTrFGFYbCDsbAOE4Afg9haLjYdC0Cux/yvSk8=";
+    rev = "v${version}";
+    sha256 = "sha256-voSbpOPJUPjwzdMLVt2TC/FIi6LKk01PLd/GczOAUR8=";
   };
 
+  nativeBuildInputs = [ poetry-core ];
   propagatedBuildInputs = [
-    docopt
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hydra-check/default.nix:25]: Still required https://github.com/nix-community/hydra-check/blob/master/pyproject.toml#L39

Comment (Artturin) [reply pkgs/development/python-modules/hydra-check/default.nix:25]: actually that doesn't mean that its required

i pushed a commit that removes that stray reference, no nixpkgs change needed

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (Artturin) [reply pkgs/top-level/all-packages.nix:17727]: i use `match case` in the program so 3.10 is required

File: pkgs/development/python-modules/hydra-check/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/hydra-check/default.nix:25]: The build failed on my end when switching my config yesterday so I am not sure here.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/hydra-check/default.nix:25]:
```
       last 10 log lines:
       > Finished executing pipBuildPhase
       > installing
       > Executing pipInstallPhase
       > /build/source/dist /build/source
       > Processing ./hydra_check-1.2.0-py2.py3-none-any.whl
       > Requirement already satisfied: requests in /nix/store/vc3ckbxix6hpyzmvlpkzkjs7w5k39hq4-python3.10-requests-2.27.1/lib/python3.10/site-packages (from hydra-check==1.2.0) (2.27.1)
       > Requirement already satisfied: beautifulsoup4 in /nix/store/kwrym5sg605kn7kfk5nd1qhkh4slv3jq-python3.10-beautifulsoup4-4.10.0/lib/python3.10/site-packages (from hydra-check==1.2.0) (4.10.0)
       > ERROR: Could not find a version that satisfies the requirement docopt (from hydra-check) (from versions: none)
       > ERROR: No matching distribution found for docopt
       >
       For full logs, run 'nix log /nix/store/6g8lgjpch17mm8h10p4jsrsmraidwfl4-python3.10-hydra-check-1.3.4.drv'.
```

Comment (Artturin) [reply pkgs/development/python-modules/hydra-check/default.nix:25]: builds for me on master
also the processing line is `python3.10-hydra-check> Processing ./hydra_check-1.3.4-py3-none-any.whl`

Comment (SuperSandro2000) [reply pkgs/development/python-modules/hydra-check/default.nix:25]: Turns out I had some old overlay which I forgot about... Sorry about the noise.

**Pull request merged**